src/pages/product-list-page/product-list-page.js

```diff
@@ -60,43 +62,61 @@ const ProductListPage = ({ category, model, width }) => {
     sortByPopularity: Products.sortByPopularity,
     countPerPage: Products.countPerPage,
     currentPage: Products.currentPage,
-    currency: Currency.currency,
     filterStatus: Products.filterStatus
   }));
-  const { categoryFilter } = filters;
+
+  const {
+    modelsFilter,
+    categoryFilter,
+    colorsFilter,
+    patternsFilter,
+    isHotItemFilter
+  } = filters;
 
   useEffect(() => {
     dispatch(getAllFilters());
   }, [dispatch]);
 
+  const fetchFilteredData = useCallback(
+    (dispatched) => {
+      console.log('callback');
+      dispatched(setCurrentPage(searchParams.get('page')));
+      dispatched(getFiltredProducts({}));
+    },
+    [
+      sortByRate,
+      sortByPrice,
+      sortByPopularity,
+      countPerPage,
+      modelsFilter.length,
+      categoryFilter.length,
+      colorsFilter.length,
+      patternsFilter.length,
+      isHotItemFilter,
+      currentPage,
+      filterStatus,
+      url
+    ]
+  );
   useEffect(() => {
-    dispatch(setCategoryFilter([category._id]));
-    dispatch(
-      setPriceFilter([
-        Math.min(
-          ...filterData.map((product) => product.basePrice[currency].value)
-        ),
-        Math.max(
-          ...filterData.map((product) => product.basePrice[currency].value)
-        )
-      ])
-    );
-  }, [category, filterData, model, currency, dispatch]);
-
-  useEffect(() => {
-    dispatch(getFiltredProducts({}));
+    fetchFilteredData(dispatch);
```

Comment:
Why do you need to set dispatch in this function

Reply:
fixed